mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -427,6 +427,13 @@ ModuleImport::processAliasScopeMetadata(const llvm::MDNode *node) {
     return node->getNumOperands() != 0 &&
            node == dyn_cast<llvm::MDNode>(node->getOperand(0));
   };
+  auto verifySelfRefOrString = [](const llvm::MDNode *node) {
+    if (node->getNumOperands() != 0 &&
+           node == dyn_cast<llvm::MDNode>(node->getOperand(0)))
+      return true;
+    return node->getNumOperands() != 0 &&
+           isa<llvm::MDNode>(node->getOperand(0));
+  };
```

Comment:
```suggestion
  // Helper that verifies the node has a self reference or a string as first operand. 
  auto verifySelfRefOrString = [](const llvm::MDNode *node) {
    if (verifySelfRef(node))
      return true;
    return node->getNumOperands() != 0 &&
           isa<llvm::MDString>(node->getOperand(0));
  };
```
Let's reuse the function above and make sure it is a string metadata node.

Reply:
The important part here is to check for MDString, currently both checks are MDNode.

Reply:
yeah sorry I failed to push the last commit [was on a plane and the wifi was ... not good ... ]